nixos/modules/system/etc/setup-etc.pl

```diff
@@ -1,52 +1,58 @@
+#! @perl@/bin/perl
+
 use strict;
-use File::Find;
-use File::Copy;
-use File::Path;
-use File::Basename;
-use File::Slurp;
+use warnings;
+
+use File::Basename qw(dirname);
+use File::Copy qw(copy);
+use File::Find qw(find);
+use File::Path qw(make_path rmtree);
+use File::Slurp qw(read_file write_file);
 
-my $etc = $ARGV[0] or die;
+my $etc = $ARGV[0] or die("This script must be called with an argument which expresses the new /etc directory in the nix store");
 my $static = "/etc/static";
 
-sub atomicSymlink {
+# Create a symlink atomically from argument 2 -> argument 1
+sub atomic_symlink {
     my ($source, $target) = @_;
     my $tmp = "$target.tmp";
-    unlink $tmp;
-    symlink $source, $tmp or return 0;
-    rename $tmp, $target or return 0;
+    unlink($tmp);
+    symlink($source, $tmp) or return 0;
+    rename($tmp, $target) or return 0;
     return 1;
 }
 
 
 # Atomically update /etc/static to point at the etc files of the
 # current configuration.
-atomicSymlink $etc, $static or die;
+atomic_symlink($etc, $static) or die("Failed to create symlink for /etc");
 
 # Returns 1 if the argument points to the files in /etc/static.  That
 # means either argument is a symlink to a file in /etc/static or a
 # directory with all children being static.
-sub isStatic {
-    my $path = shift;
+sub is_static {
+    my $path = @_;
```

Comment:
```suggestion
    my $path = shift;
```

When @_ is in scalar context, it returns the number of elements in it.

Example:
```
$ perl -E 'sub foo { my $a = @_; warn "Got $a" }; foo("file")';
Got 1 at -e line 1.
```

Reply:
just found out about this behavior in perl, thanks

Reply:
Sidenote: Perl has had support for [subroutine signatures](https://www.perl.com/article/72/2014/2/24/Perl-levels-up-with-native-subroutine-signatures/) for about 8 years now, but this feature is classified as experimental still... Hopefully that will change in 5.36.

```perl
use feature qw(signatures);
sub foo ($a, $b = "foo", @c) {
 ...
}
```

Reply:
phew, do i really want to introduce experimental features?:D

or was this purely meant as information?

Reply:
The signature feature has been stable since 5.24, seems unlikely to change, and is widely used. I wouldn't hesitate using it in production code :-) 

- https://github.com/Perl/perl5/issues/18537 

Using `use experimental qw(signatures);` instead will mute the warning btw.